chall07/estina.go

```diff
@@ -0,0 +1,48 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   estina.go                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: marvin <marvin@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/04/23 13:38:49 by marvin            #+#    #+#             */
+/*   Updated: 2020/04/23 13:38:49 by marvin           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func throw_error(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func send_get_from_server(s string) []byte {
+	resp, err := http.Get(s)
+	throw_error(err)
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	return (body)
+}
+
+func main() {
+	url := strings.Split(string(send_get_from_server("https://chall07.42madrid.com/")), " ")
+	result := strings.Split(string(url[0]), ",")
+	hexa_color := ""
+	for i := 1; i < 4; i++ {
+		color, err := strconv.Atoi(result[i][2:])
+		throw_error(err)
+		hexa_color = hexa_color + fmt.Sprintf("%x", color)
+	}

```

Comment:
A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	body := send_get_from_server("https://chall07.42madrid.com/")
	_, err := fmt.Sscanf(string(body), "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue)
	throw_error(err)
	body = send_get_from_server(fmt.Sprintf("https://chall07.42madrid.com/?id=%d&resp=%02x%02x%02x", id, red, green, blue))
	fmt.Println(string(body))
```

Reply:
It's difficult to understand the code, but I will try to reproduce that.